### PR TITLE
Implement Ties struct

### PR DESCRIPTION
**Pull request by DJBen**
Solves #42

### CODE REVIEW

File: Sources/MusicXML/Complex Types/Note.swift [hunk 1]
```diff
@@ -150,40 +139,39 @@ extension Note: Codable {
         self.notations = try container.decodeIfPresent(Notations.self, forKey: .notations)
         self.lyrics = try container.decodeIfPresent([Lyric].self, forKey: .lyrics)
         self.play = try container.decodeIfPresent(Play.self, forKey: .play)
+        let chordEmptyElement = try container.decodeIfPresent(Empty.self, forKey: .chord)
+        let tiesArray = try container.decodeIfPresent([Tie].self, forKey: .tie)
+        let ties = tiesArray == nil ? nil : Ties(ties: tiesArray!)
```
Comment (jsbean) [inline Sources/MusicXML/Complex Types/Note.swift]: Could this be a little tighter like so:

```Swift
let ties = try container.decodeIfPresent([Tie].self, forKey: .tie).map(Ties.init)
```

File: Sources/MusicXML/Complex Types/Note.swift [hunk 2]
```diff
@@ -150,40 +139,39 @@ extension Note: Codable {
...
         // Decode pitch / unpitched / rest
         let pitchUnpitchedRestContainer = try decoder.singleValueContainer()
         let pitchUnpitchedOrRest = try pitchUnpitchedRestContainer.decode(PitchUnpitchedOrRest.self)
         // Decode kind
-        do {
-            let kindContainer = try decoder.container(keyedBy: Normal.CodingKeys.self)
-            self.kind = .normal(
-                Normal(
-                    chord: kindContainer.contains(.chord),
+        if container.contains(.grace) {
```
Comment (jsbean) [inline Sources/MusicXML/Complex Types/Note.swift:148]: 🤙

File: Sources/MusicXML/Complex Types/Note.swift [hunk 3]
```diff
@@ -150,40 +139,39 @@ extension Note: Codable {
         self.notations = try container.decodeIfPresent(Notations.self, forKey: .notations)
         self.lyrics = try container.decodeIfPresent([Lyric].self, forKey: .lyrics)
         self.play = try container.decodeIfPresent(Play.self, forKey: .play)
+        let chordEmptyElement = try container.decodeIfPresent(Empty.self, forKey: .chord)
```
Comment (jsbean) [inline Sources/MusicXML/Complex Types/Note.swift:141]: This is interesting. Have you tried reenabling the test `NoteTests.DISABLED_testChord()`? We have been having `<chord/>` issues, but perhaps it was user error on my side?

cc @bwetherfield.

File: Sources/MusicXML/Complex Types/Note.swift [hunk 4]
```diff
@@ -126,6 +109,12 @@ extension Note: Codable {
...
         case lyrics
         case play
+        // Normal Note, Cue and Grace
+        case grace
+        case cue
+        case chord
+        case duration
+        case tie
     }
     #warning("Handle Ties decode once Ties struct is in place")
```
Comment (jsbean) [inline Sources/MusicXML/Complex Types/Note.swift]: I am assuming we can get rid of this `#warning` with these changes?

File: Sources/MusicXML/Complex Types/Ties.swift [hunk 5]
```diff
@@ -0,0 +1,30 @@
+//
+//  File.swift
+//  
+//
+//  Created by Ben Lu on 10/1/19.
+//
+
+import Foundation
```
Comment (jsbean) [inline Sources/MusicXML/Complex Types/Ties.swift]: Do we need `Foundation` here?

File: Sources/MusicXML/Complex Types/Note.swift (hunk 3 above)
Comment (DJBen) [reply Sources/MusicXML/Complex Types/Note.swift:141]: `testChord` errors if I reenable the test. Meanwhile this chord element is always `nil`.

**Comment by DJBen** [comment]
@jsbean Hi I updated the pull request. Would you able to take another look? Thanks

**Comment by jsbean** [comment]
Sorry, was packed today, will merge this tomorrow morning!

**Comment by jsbean** [comment]
This is great. Thanks, @DJBen!

**Pull request merged**